### DISCUSSION TITLE
What's New for Developers (February 2023)

### DISCUSSION THREAD

**Discussion by justintadlock** [WordPress/developer-blog-content#48] [Topic Ideas]
This is a rolling discussion for a new monthly series that is intended to keep track of the items, PRs, Trac ticket and make blog posts that are interesting/important for developers.  The current goal is to publish a post on the 10th of each month with the previous month's updates.

Everyone is welcome to contribute to this monthly document.  If you find something interesting, just create a new reply to this discussion below.

**Resources to track:**

- [Keeping up with Gutenberg: Index 2022](https://make.wordpress.org/core/handbook/references/keeping-up-with-gutenberg-index/) 
- [A Week in Core](https://make.wordpress.org/core/tag/week-in-core/)
- [Gutenberg plugin release posts](https://make.wordpress.org/core/tag/gutenberg-new/)
- [Core editor chat](https://make.wordpress.org/core/tag/core-editor-summary/), [dev chat](https://make.wordpress.org/core/tag/dev-chat+summary/), [performance team](https://make.wordpress.org/core/tag/performance-chat+summary/) meeting summaries

**[Google Doc (Draft)](https://docs.google.com/document/d/14eYHwo23vMqVOt4pJNrzdU83dgOtf8MTiaTt9blcn4c/edit)**

**Comment by justintadlock** [comment]
Gutenberg 14.9: Block patterns now support a new `templateTypes` option: https://github.com/WordPress/gutenberg/pull/45814

This allows theme authors to register patterns for specific template types (e.g., 404, single, etc.)

**Comment by justintadlock** [comment]
Gutenberg 14.9: `theme.json` support for custom box-shadows: https://github.com/WordPress/gutenberg/pull/46813

Related topic idea: https://github.com/WordPress/developer-blog-content/discussions/47

**Comment by bph** [comment]
[Template Parts: Add an option to import widgets from the sidebars](https://github.com/WordPress/gutenberg/pull/45509#top)  also 14.9 
Apart from the feature coming. Is there implication for a migration path from a classic theme to a block theme, or hybrids? 
Not sure if there is a developer angle to it, though. 

@noisysocks has some great insights in the workings of the Legacy Widget [in this comment](https://github.com/WordPress/gutenberg/pull/45509#issuecomment-1350404938)

**Reply by bph** [reply]
Still not sure... let's test it.

**Comment by bph** [comment]
From 14.7 I have marked a few PRs that might be suitable for Dev Blog? Executive word is **might** - I am so unsure of the ramification for developer: 

- [x] Bump DateTimePicker deprecated prop removal version. ([46006](https://github.com/WordPress/gutenberg/pull/46006)) #DevBlog
- [x] ComboboxControl: Add new opt-in prop. ([45796](https://github.com/WordPress/gutenberg/pull/45796)) #DevBlog
- [x] FocalPointPicker: Add new opt-in prop. ([45958](https://github.com/WordPress/gutenberg/pull/45958))#devBlog
- [x] RangeControl: Remove margin override and add new opt-in prop. ([45985](https://github.com/WordPress/gutenberg/pull/45985)) #DevBlog
- [x] SearchControl: Remove margin overrides and add new opt-in prop. ([46081](https://github.com/WordPress/gutenberg/pull/46081)) #DevBlog
- [x] Add Layout controls to children of Flex layout blocks. ([45364](https://github.com/WordPress/gutenberg/pull/45364))

_I added the checkboxes to mark those which actually qualify :-)_

**Comment by bph** [comment]
from 14.8 
[Add new selector getLastInsertedBlockClientId](https://github.com/WordPress/gutenberg/pull/46531)

**Comment by bph** [comment]
also 14.8 
[Query: Remove color block supports](https://github.com/WordPress/gutenberg/pull/46147)
>Removes the Query block's color supports, moving them to the inner blocks.
>Note: This could be a potentially breaking change due to the potential introduction of a wrapping Group block to enable reapplying the prior Query block colors.

**Comment by bph** [comment]
Tracking issue wrapping up Phase 2 

- https://github.com/WordPress/gutenberg/issues/33094#issuecomment-1376337709

**Comment by bph** [comment]
From Ryan's stream

- [ ] https://github.com/WordPress/gutenberg/pull/46373
- [x] https://github.com/WordPress/gutenberg/pull/45814
- [x] https://github.com/WordPress/gutenberg/pull/46559
- [x] https://github.com/WordPress/gutenberg/pull/46813
- [x] https://github.com/WordPress/gutenberg/pull/42489

**Comment by juanmaguitar** [comment]
@jonathanbossenger  is there any learning resource for developers published in the last month that you think should be highlighted to the development community?

**Comment by juanmaguitar** [comment]
@mburridge, as you're leading the release of Gutenberg 15.0, is there any specific feature for developers on this release that caught your attention?

**Comment by justintadlock** [comment]
We need to draw from the Learn WordPress workshops included here: https://www.meetup.com/learn-wordpress-online-workshops/events/past/

We can probably just list those out as a reply to this comment.  Do we want to include links to past events, assuming they are on WordPress TV or YouTube?  Do we want to include upcoming workshops?

**Reply by bph** [reply]
The links to past events are less helpful unless there were WordPressTV/Youtube recordings, then I would link to the recordings, we find [under "How to"](https://wordpress.tv/category/how-to/) Adding upcoming events are more helpful, I would think. They don't age well. But the What's new for Developers" post is more of an ephemeral post anyway. 
We do have a Learn WordPress secti
<img width="633" alt="Screenshot 2023-01-17 at 22 30 43" src="https://user-images.githubusercontent.com/39980/213016545-6d46a03a-d3b8-4826-9638-3120f199efd5.png">
on that lists all the developer related courses available and also links to the Online Workshops

**Comment by bph** [comment]
15.0 
The changes to the TabPanel seems interesting. You can now disable Tab Panels The component was updated. 

- https://github.com/WordPress/gutenberg/pull/46471
- https://github.com/WordPress/gutenberg/pull/47100
- Link to [Storybook: TabPanel](https://wordpress.github.io/gutenberg/?path=/docs/components-tabpanel--default)

**Comment by mburridge** [comment]
15.0 
- https://github.com/WordPress/gutenberg/pull/46885

**Comment by bph** [comment]
This was converted to #74 - please continue discussion there.

**Comment by bph** [comment]
Published: [What’s new for developers? (February 2023)](https://developer.wordpress.org/news/2023/02/whats-new-for-developers-february-2023/)
